tools/config/go-templates/model_simple.mustache

```diff
@@ -172,14 +172,15 @@ func (o *{{classname}}) Set{{name}}(v {{vendorExtensions.x-go-base-type}}) {
 
 {{/required}}
 {{/vars}}
-func (o {{classname}}) MarshalJSONWithoutReadOnly() ([]byte, error) {
+func (o *{{classname}}) MarshalJSONWithoutReadOnly() ([]byte, error) {
```

Comment:
this is the change, the rest is regenerating things

Reply:
i suppose we prefer this because performance?

Reply:
you need pointers to modify the struct, which we do for setters, and you should not mix pointers and non pointers so stick to the one you need

<img width="1198" alt="Screenshot 2024-10-01 at 11 53 18" src="https://github.com/user-attachments/assets/1e9a6827-4e21-44b2-970c-2ddbe21679d5">

Reply:
thanks, i didn't think of the setters